src/components/Filter.js

```diff
@@ -1,19 +1,43 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { sortMovies, fetchMovies } from '../actions/index'
 import PropTypes from 'prop-types';
 
-const Filter = ({movieCount}) => (
-    <section className="filter-container">
-        <span className="results">{movieCount} movies found</span>
-        <div className="sort-detail">
-            <span>Sort by</span>
-            <span>release date</span>
-            <span>rating</span>
-        </div>
-    </section>
-);
+class Filter extends Component {
+
+    handleSortClick = (event) => {
+        const sortBy = event.target.id === "release" ? "release_date" : "vote_average";
+        this.props.sortMovies(sortBy);
+        this.props.fetchMovies();
+
+        document.getElementById("release").classList.toggle("highlight-color");
+        document.getElementById("rating").classList.toggle("highlight-color");
```

Comment:
I'd recommend declarative (over imperative) programming style here (and whenever possible) :)

Let me suggest below refactor to show the difference. Please check inline comments for more details :)
```jsx
class Filter extends Component {
    handleSortClick = (event) => {
      const sortBy = event.target.id === "release" ? "release_date" : "vote_average";
      this.props.sortMovies(sortBy);
      this.props.fetchMovies();
    };
  
    render() {
      const { movieCount, activeSorting } = this.props;

      return (
        <section className="filter-container">
          <span className="results">{movieCount} movies found</span>
          <div className="sort-detail">
            <span>Sort by</span>
            {/* declaring the condition when highlight-color classname should be applied */}
            <span id="release" className={activeSorting === 'release_date' ? 'highlight-color' : ''} onClick={this.handleSortClick}>release date</span>
            <span id="rating" className={activeSorting === 'vote_average' ? 'highlight-color' : ''} onClick={this.handleSortClick}>rating</span>
          </div>
        </section>
       )
    }
  }
  
  Filter.propTypes = {
    movieCount: PropTypes.number,
    activeSorting: PropTypes.string, // added prop
  };
  
  const mapStateToProps = (state) => ({
    movieCount: state.movies.movieData.length,
    activeSorting: state.sortby, // get active sortby value from state
  });
  
  const mapDispatchToProps = { // object short-hand notation for mapDispatchToProps
    sortMovies,
    fetchMovies,
  }
  
  export default connect(mapStateToProps, mapDispatchToProps)(Filter); 
```